### PR TITLE
Add Jekyll project to generate the static document site

### PR DESCRIPTION
**Pull request by codingpotato**
This is an experimental PR to generate the static document site with the `Just the docs` theme. The default Github action will generate and deploy the static site from `docs` folder of `main` branch to `https://seedv.github.io/SeedLang`.

The markdown files and configurations can be modified in later PRs.

### CODE REVIEW

File: docs/Gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,84 @@
+GEM
```
Comment (wixette) [inline docs/Gemfile.lock:1]: Does the file `Gemfile.lock` need to be checked in?

When I `bundle install` in the dir, the following error message was got:

```
bundle install
Fetching gem metadata from https://rubygems.org/.........
Resolving dependencies...
Bundler could not find compatible versions for gem "jekyll-feed":
  In snapshot (Gemfile.lock):
    jekyll-feed (= 0.16.0)

  In Gemfile:
    jekyll-feed (~> 0.12)

    github-pages was resolved to 36, which depends on
      jekyll-feed (= 0.2.3)

Deleting your Gemfile.lock file and running `bundle install` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
Bundler could not find compatible versions for gem "kramdown":
  In snapshot (Gemfile.lock):
    kramdown (= 2.4.0)

  In Gemfile:
    github-pages was resolved to 8, which depends on
      kramdown (= 1.0.2)

    jekyll-feed (~> 0.12) was resolved to 0.16.0, which depends on
      jekyll (>= 3.7, < 5.0) was resolved to 4.2.2, which depends on
        kramdown (~> 2.3)

Deleting your Gemfile.lock file and running `bundle install` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```

I had to manually delete the file `Gemfile.lock` then run `bundle install` again.

File: docs/Gemfile [hunk 2]
```diff
@@ -0,0 +1,35 @@
+source "https://rubygems.org"
+# Hello! This is where you manage which Jekyll version is used to run.
```
Comment (wixette) [inline docs/Gemfile:2]: No need to include the help comments in this file I think.

**Comment by wixette** [comment]
This PR is so cool! Are we planning to merge this PR into the main branch?

**Comment by codingpotato** [comment]
> This PR is so cool! Are we planning to merge this PR into the main branch?

I plan to merge it into the main branch. The default Github action will triggered when the `docs` folder of the main branch is changed, and generate the docs in https://seedv.github.io/SeedLang/.

File: docs/Gemfile.lock (hunk 1 above)
Comment (codingpotato) [reply docs/Gemfile.lock]: `Gemfile.lock` is removed from the repo.

File: docs/Gemfile (hunk 2 above)
Comment (codingpotato) [reply docs/Gemfile]: The help comments is removed.

**Review by wixette — APPROVED** [review]

**Pull request merged**